### PR TITLE
MAKAIRA-3311 [Component] Select, AutoComplete

### PR DESCRIPTION
**Pull request by nfqphihungnguyen**

### CODE REVIEW

File: src/components/Select/Select.module.scss [hunk 1]
```diff
@@ -0,0 +1,255 @@
+@import '../../styles/mixins';
```
Comment (JanMaeg) [inline src/components/Select/Select.module.scss:1]: Please add blank lines to this file. After every block on the same level should be blank line.

File: next.config.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 /** @type {import('next').NextConfig} */
-const nextConfig = {
+const withLess = require('next-with-less')
```
Comment (JanMaeg) [inline next.config.js]: Why adding less here. Every other `rc-*` works without less. Is this really required here?

File: src/components/Select/Select.module.scss [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+
+.select-input {
+  display: inline-block;
+  :global {
+    .select-input__label {
```
Comment (JanMaeg) [inline src/components/Select/Select.module.scss]: These classes are defined by ourself so they don't need to be in the global context.
Use the global context only when overriding classes from external libraries.

File: src/components/Select/Select.tsx [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    points: ['bl', 'tl'],
+    offset: [0, -4],
+    overflow: {
+      adjustX: 0,
+      adjustY: 1,
+    },
+  },
+}
+
+const Select: React.FC<SelectProps> = ({
```
Comment (JanMaeg) [inline src/components/Select/Select.tsx:50]: There are errors in the console. Please fix them.

For example:
```
Warning: Prop `searchValue` did not match. Server: "null" Client: ""
    at svg
    at IconBase
    at span
    at TransBtn (webpack-internal:///./node_modules/rc-select/es/TransBtn.js:10:24)
    at div
    at eval (webpack-internal:///./node_modules/rc-select/es/BaseSelect.js:59:18)
    at eval (webpack-internal:///./node_modules/rc-select/es/Select.js:88:18)
    at div
    at Select (webpack-internal:///./src/components/Select/Select.tsx:54:11)
    at div
    at PageWrapper (webpack-internal:///./src/components/PageWrapper/PageWrapper.tsx:17:11)
    at Example
    at MakairaAppProvider (webpack-internal:///./src/makaira/MakairaAppProvider.tsx:22:11)
    at QueryClientProvider (webpack-internal:///./node_modules/@tanstack/react-query/build/lib/QueryClientProvider.mjs:47:3)
    at main
    at App (webpack-internal:///./src/pages/_app.tsx:31:11)
    at PathnameContextProviderAdapter (webpack-internal:///./node_modules/next/dist/shared/lib/router/adapters.js:62:11)
    at ErrorBoundary (webpack-internal:///./node_modules/next/dist/compiled/@next/react-dev-overlay/dist/client.js:301:63)
    at ReactDevOverlay (webpack-internal:///./node_modules/next/dist/compiled/@next/react-dev-overlay/dist/client.js:850:919)
    at Container (webpack-internal:///./node_modules/next/dist/client/index.js:62:1)
    at AppContainer (webpack-internal:///./node_modules/next/dist/client/index.js:172:11)
    at Root (webpack-internal:///./node_modules/next/dist/client/index.js:347:11) 

See more info here: https://nextjs.org/docs/messages/react-hydration-error
```

File: src/components/Select/Select.tsx [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+        dropdownAlign={PLACEMENTS[customDropdownPlacement]}
+        prefixCls="makaira-select"
+        dropdownRender={dropdownRender}
+        {...rest}
+      >
+        {groupOptions &&
+          groupOptions.map((group) => (
+            <OptGroup label={group.label} key={group.label}>
+              {group.children.map((option) => (
+                <Option
```
Comment (JanMaeg) [inline src/components/Select/Select.tsx:87]: The options don't use the correct font.

File: src/components/Select/Select.tsx [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+  return (
+    <div className={classNames}>
+      {title && <div className="select-input__label">{title}</div>}
+      <div className="select-input__main"></div>
+      <RcSelect
+        inputIcon={FaAngleDown}
+        clearIcon={FaTimes}
+        dropdownClassName={csx(
+          styles['select-input__dropdown'],
+          size === 'large' ? styles['select-input--large'] : ''
```
Comment (JanMaeg) [inline src/components/Select/Select.tsx]: The styling of the large option is not correct. Please take a look here and compare it. https://stage.makaira.io/pali/?path=/story/atoms-form-elements-select--large

**Review by JanMaeg — CHANGES_REQUESTED** [review]

File: src/components/Select/Select.tsx (hunk 5 above)
Comment (nfqphihungnguyen) [reply src/components/Select/Select.tsx:87]: The options panel is added out of the `main`. But the fonts are only loaded in `main`. I had some updates to add the fonts to body in other PR.

**Pull request closed without merging**